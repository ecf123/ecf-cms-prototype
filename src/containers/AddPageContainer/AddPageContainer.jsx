import "./AddPage.scss";
import { useState } from "react";
import AddPage from "../../components/AddPage/AddPage";

const AddPathway = () => {
  const [pageInfo, setPageInfo] = useState({
    file: null,
    fileName: "",
    shortInputOne: "",
    shortInputTwo: "",
    freeTypeInputOne: "",
    freeTypeInputTwo: "",
  });

  const [errorMessage, setErrorMessage] = useState("");
  const [pageFirstOpened, setPageFirstOpened] = useState(true);

  const file= pageInfo.file
  const fileName= pageInfo.fileName

  const classNameInput =
    (pageInfo.freeTypeInputOne != "" &&
    !freeTypeLabelTextOne.includes("optional") &&
    !pageFirstOpened) ||
    (pageFirstOpened && !freeTypeLabelTextOne.includes("optional")) ||
    freeTypeLabelTextOne.includes("optional")
    ? "free-type__input"
    : "free-type__input--empty";


  const classNameInputTwo = 
    (pageInfo.shortInputTwo != "" &&
    !shortLabelTextTwo.includes("optional") &&
    !pageFirstOpened) ||
    (pageFirstOpened && !shortLabelTextTwo.includes("optional")) ||
    shortLabelTextTwo.includes("optional")
    ? "short__input"
    : "short__input--empty"

  const handleFileChange = (e) => {
    if (!pageInfo.file) setPageInfo({ fileName: e.target.files[0].name });
    pageInfo.file
      ? setPageInfo({ ...pageInfo, file: null })
      : setPageInfo({
          ...pageInfo,
          file: URL.createObjectURL(e.target.files[0]),
        });
  };

  const handleChange = (e) => {
    console.log(e.target.name);
    setPageInfo({ ...pageInfo, [e.target.name]: e.target.value });
  };

  const storeInputs = (event) => {
    setPageFirstOpened(false);
    event.preventDefault();
    if (
      (pageInfo.shortInputOne == "" &&
        !shortLabelTextOne.includes("optional")) ||
      (pageInfo.freeTypeInputOne == "" &&
        !freeTypeLabelTextOne.includes("optional")) ||
      (pageInfo.shortInputTwo == "" &&
        !shortLabelTextTwo.includes("optional")) ||
      (pageInfo.freeTypeInputTwo == "" &&
        !freeTypeLabelTextTwo.includes("optional"))
    ) {
      setErrorMessage("Please enter the info needed in the empty red boxes");
    } else {
      setErrorMessage("");
    }
  };

  const propsObject = {
    uploadLabelName: uploadLabelName,
    handleFileChange: handleFileChange,
    handleChange: handleChange,
    storeInputs: storeInputs,
    errorMessage: errorMessage,
    classNameInput: classNameInput,
    classNameInputTwo: classNameInputTwo,
    file: file,
    fileName: fileName
  } 


  return (
    <div className="add-pathway">
      <AddPage shortLabelTextOne={"Pathway Name"} shortLabelTextTwo={"Pathway Category(s)"} shortPlaceHolderTextOne={""} shortPlaceHolderTextTwo={"Business"} freeTypeLabelTextOne={"Pathway Overview"} freeTypeLabelTextTwo={"Pathway Career Discription"} propsObject={propsObject}/>
    </div>
  );
};

export default AddPathway;
