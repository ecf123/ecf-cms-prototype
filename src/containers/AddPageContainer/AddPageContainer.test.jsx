import { customRender } from "../../utils/testUtils";
import AddPageContainer from "./AddPageContainer";

it("it should match the snapshot with props", () => {
    const { container } = customRender(<AddPageContainer />);
    expect(container).toMatchSnapshot();
});

  