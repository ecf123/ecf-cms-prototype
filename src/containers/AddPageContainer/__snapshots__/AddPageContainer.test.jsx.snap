// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot with props 1`] = `
<div>
  <div
    class="add-pathway"
  >
    <div
      class="add-pathway"
    >
      <div
        class="add-pathway__input-texts"
      >
        <div
          class="short"
        >
          <label
            class="short__label"
          >
            Pathway Name
          </label>
          <input
            class="short__input"
            name="shortInputOne"
            placeholder=""
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="free-type"
        >
          <label
            class="free-type__label"
          >
            Pathway Overview
          </label>
          <textarea
            class="free-type__input"
            name="freeTypeInputOne"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="short"
        >
          <label
            class="short__label"
          >
            Pathway Category(s)
          </label>
          <input
            class="short__input"
            name="shortInputTwo"
            placeholder="Business"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="free-type"
        >
          <label
            class="free-type__label"
          >
            Pathway Career Discription
          </label>
          <textarea
            class="free-type__input"
            name="freeTypeInputTwo"
            spellcheck="true"
            type="text"
          />
        </div>
        <p
          class="add-pathway__error-message"
        />
        <div
          class="add-pathway__buttons"
        >
          <div
            class="add-pathway__cancel"
          >
            <button
              class="button black round-border large"
            >
              <div />
              <p
                class="text large-text black"
              >
                Submit
              </p>
            </button>
          </div>
          <div
            class="add-pathway__submit"
          >
            <button
              class="button light-grey round-border large"
            >
              <div />
              <p
                class="text large-text light-grey"
              >
                Cancel
              </p>
            </button>
          </div>
        </div>
      </div>
      <div
        class="add-pathway__media-upload"
      >
        <div
          class="thumbnail"
        >
          <label
            class="thumbnail__label"
          >
            Pathway Thumbnail
          </label>
          <div
            class="thumbnail__container"
          >
            <div
              class="thumbnail__btn--wrapper"
            >
              <button
                class="button light-grey round-border small"
              >
                <div />
                <p
                  class="text small-text light-grey"
                >
                  Upload Image
                </p>
              </button>
              <input
                accept="image/*,video/*,audio/*,.txt"
                class="thumbnail__btn--input"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
